pkg/mcp/details.go

```diff
@@ -25,6 +26,10 @@ func (sm *SessionManager) GetServerDetails(ctx context.Context, serverConfig Ser
 
 	var deployment appsv1.Deployment
 	if err := sm.client.Get(ctx, client.ObjectKey{Name: id, Namespace: sm.mcpNamespace}, &deployment); err != nil {
+		if apierrors.IsNotFound(err) {
+			return types.MCPServerDetails{}, fmt.Errorf("mcp server %s is not running", id)
+		}
```

Comment:
This and the other change in this file are unrelated to the main goal of this PR; I just figured we might as well handle this error better.